packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx

```diff
@@ -51,6 +51,63 @@ type BaseHeaderProps = {
   | 'onPointerUp'
 >;
 
+type TrailingComponents = 'Button' | 'Badge' | 'Link' | 'Text';
+
+// prop restriction map for corresponding sub components
+const propRestrictionMap = {
+  Button: {
+    size: 'xsmall',
+    variant: 'tertiary',
+  },
+  Badge: {
+    size: 'medium',
+  },
+  Link: {
+    size: 'medium',
+  },
+  Text: {
+    size: 'medium',
+    variant: 'body',
+  },
+} as const;
+
+const useTrailingRestriction = (trailing: React.ReactNode): React.ReactNode => {
+  const [
+    validatedTrailingComponent,
+    setValidatedTrailingComponent,
+  ] = React.useState<React.ReactElement | null>(null);
+
+  // validate and restrict sub component props in trailing prop
+  React.useEffect(() => {
+    if (React.isValidElement(trailing)) {
+      const trailingComponentType = getComponentId(trailing) as TrailingComponents;
+      const restrictedProps = propRestrictionMap[trailingComponentType];
+      const allowedComponents = Object.keys(propRestrictionMap);
+      if (!restrictedProps) {
+        throw new Error(
+          `[Blade Header]: Only one of \`${allowedComponents.join(
+            ', ',
+          )}\` component is accepted as trailing`,
+        );
+      }
+
+      const restrictedPropKeys = Object.keys(propRestrictionMap[trailingComponentType]);
+      for (const prop of restrictedPropKeys) {
+        if (trailing?.props?.hasOwnProperty(prop)) {
+          console.warn(
+            `[Blade BottomSheetHeader]: Do not pass "${prop}" to "${trailingComponentType}" while inside BottomSheetHeader trailing, because we override it.`,
```

Comment:
Let's use metaComponentName for the warning.

Reply:
Making it `Header`, metaComponentName will mostly be kebab case with smallcase letters. Will look weird.